### PR TITLE
User can view articles feature

### PR DESCRIPTION
**Pull request by NiciP**
#### PT Story: [https://www.pivotaltracker.com/story/show/164490799]

_As a Visitor
In order to choose an article to read
I would like to be able to see the articles listed on a page_

#### Description
- Add index page to the application
- List articles with title & lead
- Add `Article` model with unit tests
- Add and config article controller
- Add feature test to see title and lead

#### What I have learned working on this feature: 
That we do not need to follow the logic of a journalist creating an article first in order for it to be viewed, it can be created in the database.

#### Screenshots
image

image

image

### CODE REVIEW

File: app/views/articles/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+<%= form_with scope: :article, url: articles_path, local: true do |form| %>
+  <p>
+    <%= form.label :title %><br>
+    <%= form.text_field :title %>
+  </p>
+
```
Comment (Joberg95) [inline app/views/articles/new.html.erb:7]: Maybe we need to add a "lead" form.label as well

File: db/migrate/20190313192026_create_articles.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateArticles < ActiveRecord::Migration[5.2]
+  def change
+    create_table :articles do |t|
+      t.string :title
+      t.text :content
+
```
Comment (Joberg95) [inline db/migrate/20190313192026_create_articles.rb:6]: will look better without this space

File: db/schema.rb [hunk 3]
```diff
@@ -10,9 +10,17 @@
...
   enable_extension "plpgsql"
 
+  create_table "articles", force: :cascade do |t|
+    t.string "title"
+    t.text "content"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.text "lead"
+  end
+
```
Comment (Joberg95) [inline db/schema.rb:25]: remove space

File: features/visitor_can_see_articles_on_index_page.feature [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+    As a Visitor
+    In order to choose an article to read
+    I would like to be able to see the articles listed on a page
+
+Background: 
+    Given the following articles exist
+            |                       title                            |                          lead                             |                                   content                                        |
+            |           Voted best mead recipe        |                Restaurant wins prize            |     Restaurant wins prize for best mead in Sweden       | 
+            |   Ancient viking grave discovered   |               Kids come across sword         |   Kids come across sword protruding from the earth    |
```
Comment (Joberg95) [inline features/visitor_can_see_articles_on_index_page.feature:11]: formatting

File: features/step_definitions/basic_steps.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+When("I visit the site") do
+    visit root_path
+end
+
+Given("the following articles exist") do |table|
+    table.hashes.each do |article|
+        Article.create!(article)
```
Comment (magnus-thor) [inline features/step_definitions/basic_steps.rb:7]: use factorybot
`create(:article, article)`

File: app/controllers/articles_controller.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
+class ArticlesController < ApplicationController
+    def index
+        @articles = Article.all
+    end
+
+    def show
```
Comment (magnus-thor) [inline app/controllers/articles_controller.rb:6]: remove, out of scope

File: app/controllers/articles_controller.rb [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        @article = Article.find(params[:id])
+    end
+
+    def new
```
Comment (magnus-thor) [inline app/controllers/articles_controller.rb:10]: remove, out of scope

File: app/controllers/articles_controller.rb [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    end   
+
+    def create
```
Comment (magnus-thor) [inline app/controllers/articles_controller.rb:13]: remove, out of scope

File: app/controllers/articles_controller.rb [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+        @article = Article.new(article_params)
+   
+        @article.save
+        redirect_to root_path
+    end
+   
+    private
```
Comment (magnus-thor) [inline app/controllers/articles_controller.rb:20]: remove, out of scope

**Review by magnus-thor — CHANGES_REQUESTED** [review]

**Comment by NiciP** [comment]
Opened new pull request for this feature

**Pull request closed without merging**